lib/mayaUsd/fileio/translators/translatorMayaReference.cpp

```diff
@@ -173,6 +189,60 @@ const TfToken UsdMayaTranslatorMayaReference::m_referenceName = TfToken("mayaRef
 const TfToken UsdMayaTranslatorMayaReference::m_mergeNamespacesOnClash
     = TfToken("mergeNamespacesOnClash");
 
+// Get required namespace attribute from prim
+MString UsdMayaTranslatorMayaReference::namespaceFromPrim(const UsdPrim& prim)
+{
+    std::string ns;
+    if (UsdAttribute namespaceAttribute = prim.GetAttribute(m_namespaceName)) {
+        TF_DEBUG(PXRUSDMAYA_TRANSLATORS)
+            .Msg(
+                "MayaReferenceLogic::update Checking namespace on prim \"%s\".\n",
+                prim.GetPath().GetText());
+
+        if (!namespaceAttribute.Get<std::string>(&ns)) {
+            TF_DEBUG(PXRUSDMAYA_TRANSLATORS)
+                .Msg(
+                    "MayaReferenceLogic::update Missing namespace on prim \"%s\". Will create one "
+                    "from prim path.\n",
+                    prim.GetPath().GetText());
+            // Creating default namespace from prim path. Converts /a/b/c to a_b_c.
+            ns = prim.GetPath().GetString();
+            std::replace(ns.begin() + 1, ns.end(), '/', '_');
+        }
+    }
+    return MString(ns.c_str(), ns.size());
+}
+
+MString UsdMayaTranslatorMayaReference::getUniqueRefNodeName(
+    const UsdPrim&      prim,
+    const MFnDagNode&   parentDag,
+    const MFnReference& refDependNode)
+{
+    MString uniqueRefNodeName;
+    if (isMayaReference(prim)) {
+        // Rename the reference node. Append "RN" to the end of filename, to indicate it's a
+        // reference node.
+        //
+        MString               filePath = refDependNode.fileName(false, false, false);
+        ghc::filesystem::path fsFilePath(filePath.asWChar());
+        fsFilePath.replace_extension(""); // Remove the extension
+        uniqueRefNodeName = namespaceFromPrim(prim);
+
+        if (uniqueRefNodeName.length() == 0) {
+            uniqueRefNodeName += fsFilePath.filename().wstring().c_str();
+        }
+        uniqueRefNodeName += L"RN";
+    } else {
```

Comment:
For my benefit, can you explain when this else clause is taken?  For AL reference prim?  It might be a good code comment.

Reply:
Why is there a difference between Maya reference prims and others?  Why can't the behavior be the same?  Is it for backward compatibility only?

Reply:
Yes, we don't want to break existing pipelines.